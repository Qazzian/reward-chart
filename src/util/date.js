export const dateFormatOptions = {
  year: 'numeric',
  month: '2-diget',
  day: '2-diget',
}

export function toDateString(date) {
	if (date instanceof Date) {
		return date.toLocaleDateString(dateFormatOptions);
	}
	else {
		throw new Error(`expected Date object but got ${date}`)
	}
}