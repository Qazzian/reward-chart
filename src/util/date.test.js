import * as dateUtil from './date'; 

describe('Date Utils', () => {
	
	it('should compile correctly', () => {
		expect(dateUtil).toBeDefined();
	});

	describe('toDateString', () => {
		const {toDateString} = dateUtil;

		it('should format the date correctly', () => {
			expect(toDateString(new Date('2018-12-14'))).toBe('2018-12-14');
		});
	});

	describe('subtractDays', () => {
		const {subtractDays} = dateUtil;
		const tests = [
			{
				name: 'calc two days off',
				startDate: '2018-12-14',
				subtract: 2,
				expected: '2018-12-12',
			},
			{
				name: 'cross month boundries',
				startDate: '2018-12-05',
				subtract: 7,
				expected: '2018-11-28',
			},
			{
				name: 'Cross year boundries',
				startDate: '2019-01-5',
				subtract: 7,
				expected: '2018-12-29',
			},
		];

		tests.forEach((test) => {
			it(`should ${test.name}`, () => {
				const startDate = new Date(test.startDate);
				const expectedDate = new Date(test.expected);
				expect(subtractDays(startDate, test.subtract).toDateString())
					.toMatch(expectedDate.toDateString());
			});
		});

	});

	describe('loopOverDays', () => {
		const {loopOverDays} = dateUtil;

		it('should iterate over the days', () => {
			const spy = jest.fn((dateParam) => {});
			loopOverDays(5, spy, new Date('2018-12-14'));
			expect(spy.mock.calls.length).toBe(5);
			expect(spy.mock.calls).toMatchSnapshot();
		});
	});
});