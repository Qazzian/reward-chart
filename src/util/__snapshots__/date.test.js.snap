// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Utils loopOverDays should iterate over the days 1`] = `
Array [
  Array [
    2018-12-10T00:00:00.000Z,
  ],
  Array [
    2018-12-11T00:00:00.000Z,
  ],
  Array [
    2018-12-12T00:00:00.000Z,
  ],
  Array [
    2018-12-13T00:00:00.000Z,
  ],
  Array [
    2018-12-14T00:00:00.000Z,
  ],
]
`;
